src/model/encoding/convertFromHTMLToContentBlocks.js

```diff
@@ -404,32 +427,37 @@ function genFragment(
     lastList = nodeName;
   }
 
+  const blockType = getBlockTypeForTag(nodeName, lastList, blockRenderMap);
+
   // Block Tags
-  if (!inBlock && blockTags.indexOf(nodeName) !== -1) {
-    chunk = getBlockDividerChunk(
-      getBlockTypeForTag(nodeName, lastList, blockRenderMap),
-      depth,
-    );
+  if (lastList && inBlock === 'li' && nodeName === 'li') {
+    chunk = getBlockDividerChunk(blockType, depth, parentKey);
+    blockKey = chunk.blocks[0].key;
```

Comment:
We get a reference of the generated `blockKey` so that wen we recurse we can pass it as the `parentKey` for the next block